lib/batch_loader.rb

```diff
@@ -36,7 +36,10 @@ def batch(default_value: nil, cache: true, key: nil, &batch_block)
   end
 
   def respond_to?(method_name, include_private = false)
-    LEFT_INSTANCE_METHODS.include?(method_name) || method_missing(:respond_to?, method_name, include_private)
+    return true if LEFT_INSTANCE_METHODS.include?(method_name)
+
+    loaded_value = defined?(@loaded_value) ? @loaded_value : __sync!
```

Comment:
>>>
I think that method_missing(:respond_to?, method_name, include_private) doesn't call the original method_missing, it calls respond_to?(method_name, include_private) through public_send? Or there is another issue which can be reproduced with a test?
>>>

Fair enough, I'll write a test to show what happens.

Reply:
@exAspArk I'll update this branch in a bit to have a failing test case. The first commit will be the failing test case, where the last commit will rectify this.

On the `__sync!` idea, we can't call `:respond_to?` on it, given it will keep recursively calling itself and thus fail when the stack gets too deep.

I've checked out if I can refactor to create a `loaded_value` method. But first wanted to get your view on it, because you have a better idea how this library works. 😄